test/test_graph.jl

```diff
@@ -0,0 +1,56 @@
+## Test sparse
+#  workspace()
+# using ProximalOperators
+#
+# using Base.Test
+
+srand(0)
+
+m, n = (10, 50)
+
+A = sprand(m, n, 0.3)
+Adf = full(A)
+B = [A -speye(m)]
+Bd = full(B)
+
+
+
+fiag = ProximalOperators.IndGraph(A)
+fiaf = ProximalOperators.IndAffine(B, zeros(m))
+fiagdf = ProximalOperators.IndGraph(Adf)
+
+c = randn(n) * 10 + 10
+d = A * c
+
+@test fiag(c, d) == 0.0
+@test fiagdf(c, d) == 0.0
+
+y = zeros(d)
+x = zeros(c)
+
+xy = [x ; y]
+cd = [c ; d]
+
+x_a = @view xy[1:n]
+y_a = @view xy[n+1:end]
+
+## IndAffine QR
+prox!(xy, fiaf, cd)
+@test fiaf(xy) == 0.0
+
+## SPARSE
+# test call
+prox!(x, y, fiag, c, d)
```

Comment:
Yes, what I'm suggesting is for example to use the `prox_test` (with `xy` and `cd` as inputs) and `call_test` also here: this will do all the checks, compare with the `_naive` implementations. All this will be done by at some point calling the 5-arguments `prox!(x, y, f, c, d)`, which is what is tested here now.

The reason for this is: I would like to clean up at some point the way tests are performed. The `test_calls` and `test_results` scripts are getting a bit messy and difficult to handle. I would rather, in the future, have one test script per function constructor, which uses `prox_test` and `call_test` for all routine checks, and then checks anything one may want to check: optimality condition of the computed proximal point, hand-curated cases with known expected outputs, equivalence with other functions (such as in this case, equivalence between `IndGraph` and `IndAffine` when appropriately used), and so on.

Therefore, say tomorrow we comment out `test_calls.jl` and `test_results.jl`, this remaining test script for `IndGraph` will not cover the call `prox!(xy, f, cd)`, but only the `prox!(x, y, f, c, d)`.